entry.go

```diff
@@ -197,27 +210,167 @@ func (te *TunaEntry) StartReverse(stream *smux.Stream) error {
 		te.close()
 		return err
 	}
+	session, err := te.getSession(false)
+
+	bytesIn := make([]uint64, 256)
+	bytesOut := make([]uint64, 256)
+
+	claimInterval := time.Duration(te.config.ClaimInterval) * time.Second
+	onErr := nknsdk.NewOnError(1, nil)
+	var npc *nknsdk.NanoPayClaimer
+	lastComputed := common.Fixed64(0)
+	lastClaimed := common.Fixed64(0)
+	lastUpdate := time.Now()
+	isClosed := false
+
+	npc, err = te.wallet.NewNanoPayClaimer(int32(claimInterval/time.Millisecond), onErr, te.config.BeneficiaryAddr)
+	if err != nil {
+		log.Fatalln(err)
+	}
 
 	go func() {
-		session, err := te.getSession(false)
-		if err != nil {
-			return
-		}
-		stream, err := session.OpenStream()
-		if err != nil {
-			return
-		}
-		stream.Close()
 		for {
-			_, err = session.AcceptStream()
+			err := <-onErr.C
 			if err != nil {
-				log.Println("Close connection:", err)
-				te.close()
-				return
+				log.Println("Couldn't claim nano pay:", err)
+				if npc.IsClosed() {
+					Close(session)
+					Close(conn)
+					isClosed = true
+					break
+				}
 			}
 		}
 	}()
 
+	go func() {
```

Comment:
This goroutine is duplicate of exit.go. We need to combine the logic and remove redundant code